src/ruby/ext/grpc/rb_channel.c

```diff
@@ -261,6 +265,65 @@ static VALUE grpc_rb_channel_init(int argc, VALUE *argv, VALUE self) {
   return self;
 }
 
+/*
+  call-seq:
+    ch = Channel.insecure_create_from_fd("fd:42", 42, {'arg1': 'value1'})
+
+  Creates channel instances from an existing file descriptor. */
+static VALUE grpc_rb_channel_insecure_create_from_fd(
+    VALUE klass, VALUE rb_target, VALUE rb_fd, VALUE rb_args) {
+#ifdef GPR_SUPPORT_CHANNELS_FROM_FD
+  VALUE self;
+  grpc_rb_channel *wrapper = NULL;
```

Comment:
I am trying to think of a way to avoid duplicating a lot of this logic. I'm not sure just yet but may get back to you about that